src/components/ha-icon.ts

```diff
@@ -14,6 +14,7 @@ import "./ha-svg-icon";
 import { debounce } from "../common/util/debounce";
 import { iconMetadata } from "../resources/icon-metadata";
 import { IconMeta } from "../types";
+import { customIconsets, CustomIcons } from "../data/custom_iconsets";
```

Comment:
This is good. Can we move all the other icon loading stuff into `data/iconsets` ?